File: my_exemplars/my_exemplar_2.c
#include<stdio.h>
#include<stdlib.h>
#include<time.h>
void simple_recursion(int my_rand){
    if (my_rand > 1){
        simple_recursion(my_rand/2);
    }
    else {
        return;
    }
}
int main(){
    srand(time(NULL));
    int my_rand = rand();
    if ((my_rand % 2) == 0){
        simple_recursion(my_rand);
    }
    else {
        simple_recursion(my_rand-1);
    }
}

File: my_exemplars/my_exemplar_2.py
import random
def simple_recursion(my_rand):
    if my_rand > 1:
        simple_recursion(my_rand // 2)
if __name__ == "__main__":
    random.seed()
    my_rand = random.randint(0, 2**31 - 1)
    if my_rand % 2 == 0:
        simple_recursion(my_rand)
    else:
        simple_recursion(my_rand - 1)